entrypoint.sh

```diff
@@ -138,7 +147,62 @@ function run_scorecard() {
   export ENABLE_SARIF=1
   export SCORECARD_BIN=/opt/goat/bin/scorecard
 
-  $SCORECARD_BIN version
+  # Pseudo Code Tasks 1 & 2 (used {} because we dont have env variables in our script for first two)
+  if ! [[ -z "${INPUT_SCORECARD_POLICY_FILE:-}" ]]; then
+    export SCORECARD_POLICY_FILE=${GITHUB_WORKSPACE:-/goat}/$INPUT_SCORECARD_POLICY_FILE
+  else
+    export SCORECARD_POLICY_FILE="/etc/opt/goat/seiso_scorecard_policy.yml"  
+  fi
+
+  if [[ -z $GITHUB_AUTH_TOKEN ]]; then
+    echo "Please provide a personal access token for you repo with the correct access."
+    return
+  fi
+
+  status_code=$(curl -s -H "Authorization: Bearer ${GITHUB_AUTH_TOKEN}" https://api.github.com/repos/"$GITHUB_REPOSITORY" -o repo_info.json -w '%{http_code}')
```

Comment:
Might make sense to take some of the comment contents from the scorecard implementation to document why this needs to be done instead of using some of the built-in runner environment variables.